src/party_menu.c

```diff
@@ -0,0 +1,2723 @@
+        } else {
+            ReadMsgDataIntoString(partyMenu->msgData, msg_0300_00085, partyMenu->unformattedStrBuf);
+            BufferItemName(partyMenu->msgFormat, 1, oldItemId);
+            BufferItemName(partyMenu->msgFormat, 2, newItemId);
+            StringExpandPlaceholders(partyMenu->msgFormat, partyMenu->formattedStrBuf, partyMenu->unformattedStrBuf);
+            if (newItemId != ITEM_GRISEOUS_ORB && oldItemId == ITEM_GRISEOUS_ORB && giratinaResult != -1) {
+                result = PARTY_MENU_STATE_PRINT_GIVE_GRISEOUS_ORB_MESSAGE;
+            } else if (newItemId == ITEM_GRISEOUS_ORB && oldItemId == ITEM_GRISEOUS_ORB) {
+                result = PARTY_MENU_STATE_PRINT_ITEM_SWAP_MESSAGE;
+            }
+        }
+        FillWindowPixelBuffer(&partyMenu->windows[PARTY_MENU_WINDOW_ID_34], 15);
+        PartyMenu_PrintBufferedMessageOnWindow34(partyMenu);
+        return result;
+    }
+    case YESNORESPONSE_NO:
+        YesNoPrompt_Destroy(partyMenu->yesNoPrompt);
+        PartyMenu_DisableMainScreenBlend_AfterYesNo();
+        return PartyMenu_SwitchItemsDeclined(partyMenu);
+    }
+
+    return PARTY_MENU_STATE_YESNO_ASK_SWITCH_ITEMS;
+}
+
+static int PartyMenu_SwitchItemsDeclined(PartyMenuStruct *partyMenu) {
+    if (partyMenu->args->context == PARTY_MENU_CONTEXT_10) {
+        ClearFrameAndWindow2(&partyMenu->windows[PARTY_MENU_WINDOW_ID_34], TRUE);
+        PartyMenu_PrintMessageOnWindow32(partyMenu, msg_0300_00029, TRUE);
+        thunk_Sprite_SetPalIndex(partyMenu->sprites[PARTY_MENU_SPRITE_ID_CURSOR], 0);
+        partyMenu->args->context = PARTY_MENU_CONTEXT_0;
+        return PARTY_MENU_STATE_1;
+    } else {
+        partyMenu->args->selectedAction = PARTY_MENU_ACTION_RETURN_10;
+        return PARTY_MENU_STATE_BEGIN_EXIT;
+    }
+}
+
+static int sub_0207C908(PartyMenuStruct *partyMenu) {
+    Pokemon *mon = Party_GetMonByIndex(partyMenu->args->party, partyMenu->partyMonIndex);
+    s32 giratinaResult;
+    int newItem = partyMenu->args->itemId;
+    int oldItem = partyMenu->monsDrawState[partyMenu->partyMonIndex].heldItem;
+    int manipulateItemResult = sub_0207C5D4(partyMenu, mon, &giratinaResult);
+    if (oldItem == ITEM_GRISEOUS_ORB && manipulateItemResult == PARTY_MENU_STATE_PRINT_ITEM_SWAP_MESSAGE && giratinaResult == 0) {
+        manipulateItemResult = PARTY_MENU_STATE_PRINT_GIVE_GRISEOUS_ORB_MESSAGE;
+    }
+    if (oldItem == ITEM_NONE) {
+        ReadMsgDataIntoString(partyMenu->msgData, msg_0300_00107, partyMenu->unformattedStrBuf);
+        BufferBoxMonNickname(partyMenu->msgFormat, 0, Mon_GetBoxMon(mon));
+        BufferItemName(partyMenu->msgFormat, 1, partyMenu->args->itemId);
+        StringExpandPlaceholders(partyMenu->msgFormat, partyMenu->formattedStrBuf, partyMenu->unformattedStrBuf);
+    } else {
+        Bag_AddItem(partyMenu->args->bag, oldItem, 1, HEAP_ID_PARTY_MENU);
+        ReadMsgDataIntoString(partyMenu->msgData, msg_0300_00085, partyMenu->unformattedStrBuf);
```

Comment:
Comment what messages read above pls (I prolly missed a ton this file so blanket comment for party_menu.c)

Reply:
Most of these are just {PLACEHOLDER} so nothing really interesting here. If people want to look them up, they can cross-reference the GMM. Commenting what the messages say inside the C code creates an opportunity for desync.